metabase/metabase
# feat(sdk): theme option to customize popover's z-index

Closes https://github.com/metabase/metabase/issues/45587

### Description

The popover for click actions are being shown behind the user’s modal. This only happens when the user puts an interactive question in a modal component of their choice, and the modal has a higher z-index than the Tippy popover.

PS. This is only an issue with the legacy Tippy popover, as Mantine-based Popovers already renders to `EMBEDDING_SDK_ROOT_ELEMENT_ID` and thus does not suffer from a z-index problem. With that said, this PR applies it to Mantine-based popover as well for consistency. The z-index value should apply to all popovers.

This PR exposes a theme option to customize the popover's z-index for all popovers.

![Arc 2567-07-16 05 20 16](https://github.com/user-attachments/assets/586965f3-d65b-41b4-8135-94eb8422dc2c)

### How to verify

- Go to https://github.com/metabase/shoppy/pull/54
- Click action popovers should render over the portal

### Demo

![image](https://github.com/user-attachments/assets/25df6aa6-e42f-4048-a5ee-faa448b4e64f)

### Checklist
